File: e3_CountLargestGroup/Problem/Solution.java
package e3_CountLargestGroup.Problem;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int countLargestGroup(int n) {

        Map<Integer,Integer> count = new HashMap<>();
        for(int i =1; i<= n; i++){
            int soma =0;
            int num = i;
            while(num != 0){
                soma+= num%10;
                num/=10;
            }
            count.put(soma,count.getOrDefault(soma,0)+1);
        }
        int groups=0;
        int maxvalue = Collections.max(count.values());
        for(int c : count.values())if(c == maxvalue)groups++;
        return groups;
    }
}

File: e3_CountLargestGroup/Problem/typescript/Solution.ts
class Solution {
    countLargestGroup(n: number): number {
        const count = new Map<number, number>();
        for (let i = 1; i <= n; i++) {
            let sum = 0;
            let num = i;
            while (num !== 0) {
                sum += num % 10;
                num = Math.floor(num / 10);
            }
            count.set(sum, (count.get(sum) || 0) + 1);
        }

        if (count.size === 0) {
            return 0;
        }

        let maxValue = 0;
        for (const value of count.values()) {
            if (value > maxValue) {
                maxValue = value;
            }
        }

        let groups = 0;
        for (const value of count.values()) {
            if (value === maxValue) {
                groups++;
            }
        }
        return groups;
    }
}

export { Solution };